File: mobile/lib/src/core/data/models/city_detail_model/city.dart
import 'package:freezed_annotation/freezed_annotation.dart';
part 'city.freezed.dart';
part 'city.g.dart';

@freezed
class City with _$City {
  factory City({
    int? id,
    String? name,
    @JsonKey(name: 'created_at') String? createdAt,
    @JsonKey(name: 'updated_at') DateTime? updatedAt,
    @JsonKey(name: 'deleted_at') dynamic deletedAt,
    double? latitude,
    double? longitude,
  }) = _City;


  factory City.fromJson(Map<String, dynamic> json) => _$CityFromJson(json);
}

File: backend/model/city.go
package model

import (
	"time"

	"gorm.io/gorm"
)

type City struct {
	ID          int            `json:"id"`
	Name        string         `json:"name"`
	CreatedAt   time.Time      `json:"created_at"`
	UpdatedAt   time.Time      `json:"updated_at"`
	DeletedAt   gorm.DeletedAt `json:"deleted_at,omitempty" gorm:"index" swaggertype:"string"`
	Latitude    *float64       `json:"latitude"`
	Longitude   *float64       `json:"longitude"`
	Weight      int            `json:"weight"`
	Region_name string         `json:"region_name"`
	Region_id   int            `json:"region_id"`
}

type FilterCity struct {
}
